### PR TITLE
feature: read social links from json file

### LINKED ISSUE #30: [FEATURE] Implement a JS function which reads the Social Links JSON file
# Implement a JS function which reads the Social Links JSON file

## Description
Implement a JS function in [`index.js`](https://github.com/Param302/SocialRepo/blob/main/src/code/index.js) which reads the JSON file [`social-links.json`](https://github.com/Param302/SocialRepo/blob/main/src/social-links.json) and return a object having keys as names of social media and links as their respective values.

## Requirements
- The JS function should read the JSON file properly.
- Implement the JS function inside [`index.js`](https://github.com/Param302/SocialRepo/blob/main/src/code/index.js) only.
- Make sure it won't break any other functionality.
- It should ignore the first key-value pair i.e. `"social_media_name": "link"`.
- Must follow the [**Code Guidelines**](https://github.com/Param302/SocialRepo/blob/main/.github/docs/README.md#code-guidelines) provided.
- To test the code, setup the extension by following [these](https://github.com/Param302/SocialRepo/blob/main/.github/docs/CONTRIBUTING.md#extension-setup) steps.

If you have any suggestion or idea, you're welcome to share.

_Let's read all the links for **SocialRepo** extension_ 🙌

### PR DESCRIPTION
**Pull request by RenzoRomeo**
# Title
Read social links from json file.

## Description
Implement a function inside ´index.js´ that reads the contents of ´social-links.json´ and returns a JS object with all the key-value pairs in the file, except for the first one.

---
## Issue Ticket Number
Fixes #30

The feature was tested using Chrome.

### CODE REVIEW

File: src/code/index.js [hunk 1]
```diff
@@ -44,3 +44,27 @@ showBtn.addEventListener("click", () => {
         table.classList.add("hidden");
     }
 });
+
+function readSocialLinks() {
+    const socialLinks = {};
+
+    fetch("../social-links.json")
```
Comment (Param302) [inline src/code/index.js:51]: return the Promise from the function from here so that I can use the data

File: src/code/index.js [hunk 2]
```diff
@@ -44,3 +44,27 @@ showBtn.addEventListener("click", () => {
...
+            });
+
+            console.log(socialLinks);
+            return socialLinks;
+        })
+        .catch(error => {
+            console.error("Error reading social links file:", error);
+            return {};
+        });
+}
```
Comment (Param302) [inline src/code/index.js:70]: Also, provide me an example of a function which is calling this function and printing all the keys and values fetched in console.

**Review by Param302 — CHANGES_REQUESTED** [review]
Hey @RenzoRomeo ,
First of all thanks for your contribution!

I have provided the review for the changes, please do the changes then I will merge the PR. Rest is all fine.

Also, make sure to follow basic guidelines, by discussion and asking to assigned the issue before working on.
As you've raised the PR, and seeing your effort I am assigning you the issue #30 .

_Happy Contributing!_

File: src/code/index.js (hunk 2 above)
Comment (RenzoRomeo) [reply src/code/index.js:70]: Sure!

This function would be used like so:
```javascript
function printSocialLinks() {
    readSocialLinks().then(socialLinks => {
        Object.entries(socialLinks).forEach(([key, value]) => {
            console.log(`${key}: ${value}`);
        });
    });
}

printSocialLinks();
```

File: src/code/index.js (hunk 1 above)
Comment (RenzoRomeo) [reply src/code/index.js]: Already fixed, my bad.

**Comment by Param302** [comment]
Hey @RenzoRomeo ,

Thanks for your valuable contribution to **SocialRepo**.

I am merging it to main branch

Looking forward for more contributions. :D
_Keep Contributing!_ 🙌

**Pull request merged**